bpm/bonita-core/bonita-process-engine/src/test/java/org/bonitasoft/engine/service/ModelConvertorTest.java

```diff
@@ -546,4 +552,71 @@ public void convert_null_business_data() {
         assertThat(businessDataReference).isNull();
     }
 
+    @Test
+    public void toArchivedFlownodeInstance_should_convert_reachStateDate_for_gateways() throws Exception {
+        final FlowNodeStateManager flowNodeStateManager = mock(FlowNodeStateManager.class);
+        doReturn(mock(FlowNodeState.class)).when(flowNodeStateManager).getState(anyInt());
+
+        assertThat(ModelConvertor.toArchivedFlowNodeInstance(new SAGatewayInstanceImpl(), flowNodeStateManager).getReachedStateDate()).isNotNull();
```

Comment:
only not null check?maybe it's better to check at least once a value with reachedStateDate != lastUpdateDate